src/main/java/org/candlepin/async/impl/ActiveMQSessionFactory.java

```diff
@@ -163,10 +168,13 @@ protected synchronized SessionManager getIngressSessionManager() throws Exceptio
     /**
      * Fetches the egress session manager, creating a new instance as necessary.
      *
+     * @throws Exception
+     *  if unable to create a {@link ServerLocator} with the configured broker URL
+     *
      * @return
      *  a SessionManager for sessions configured for sending messages
      */
-    protected synchronized SessionManager getEgressSessionManager() throws Exception {
+    protected SessionManager getEgressSessionManager() throws Exception {
```

Comment:
Wait, why did we remove the synchronized modifier from this?

Reply:
I thought the method wasn't used by multiple threads, but it is. Added the synchronized modifier back!

Reply:
For future reference, even if your initial assessment were accurate, we still wouldn't want to remove the synchronized modifier from this method as its design is dependent on it for (future) multi-threaded behavior; Especially given that one method was to remain thread-safe. The way that method is implemented, if multiple threads call into it at roughly the same time, we may end up with multiple session managers.

Reply:
That makes sense!